src/registrar/models/domain.py

```diff
@@ -960,82 +960,85 @@ def _fetch_cache(self, fetch_hosts=False, fetch_contacts=False):
             # statuses can just be a list no need to keep the epp object
             if "statuses" in cleaned.keys():
                 cleaned["statuses"] = [status.state for status in cleaned["statuses"]]
+
+            # Capture and store old hosts and contacts from cache if they exist
+            old_cache_hosts = self._cache.get("hosts")
+            old_cache_contacts = self._cache.get("contacts")
+
             # get contact info, if there are any
             if (
-                # fetch_contacts and
-                "_contacts" in cleaned
+                fetch_contacts
```

Comment:
QQ: Could you explain what this check is doing (still absorbing EPP knowledge 😅)?

Otherwise all the other code looks pretty great from my limited EPP knowledge, great work!

Reply:
If contacts are passed as the requested arg (that's the cache optimization), _contacts exist in the 'thin' returned object from DomainInfo (called in _get_or_create_domain) and are a list that contains at least one item, fetch the full contact objects. @abroddrick or @zandercymatics can weigh in I missed anything.